### PR TITLE
Add S3 assets download and extraction

### PR DESCRIPTION
**Pull request by andytson-inviqa**

### CODE REVIEW

**Comment by andytson-inviqa** [comment]
I'll be adding php-apache versions once this is discussed

File: php-nginx/README.md [hunk 1]
```diff
@@ -96,6 +96,15 @@ WEB_HTTPS_ONLY      | Whether to redirect all HTTP traffic to HTTPS | true/false
 WEB_REVERSE_PROXIED | Whether to interpret X-Forwarded-Proto as the $custom_scheme and $custom_https emulation. | true/false | true
 WEB_SSL_FULLCHAIN | The location of the SSL certificate and intermediate chain file | absolute filename | /etc/ssl/certs/fullchain.pem
 WEB_SSL_PRIVKEY | The location of the SSL private key file | absolute filename | /etc/ssl/private/privkey.pem
+ASSETS_ENV | The assets environment assets are downloaded/applied from. If unset no asset actions will happen | a asset environment | 
+ASSETS_PATH | The assets filesystem path assets are downloaded to /applied from. If unset and ASSETS_ENV unset no asset actions will happen | a asset filesystem path | tools/assets/${ASSETS_ENV} if ASSETS_ENV set
+ASSETS_S3_BUCKET | The AWS S3 bucket assets are downloaded from. If unset, no assets will be downloaded | a S3 bucket name |
```
Comment (sroze) [inline php-nginx/README.md]: Can't we just have `ASSETS_S3_BUCKET_PATH ` for the same reason ?

File: php-nginx/README.md [hunk 2]
```diff
@@ -96,6 +96,15 @@ WEB_HTTPS_ONLY      | Whether to redirect all HTTP traffic to HTTPS | true/false
 WEB_REVERSE_PROXIED | Whether to interpret X-Forwarded-Proto as the $custom_scheme and $custom_https emulation. | true/false | true
 WEB_SSL_FULLCHAIN | The location of the SSL certificate and intermediate chain file | absolute filename | /etc/ssl/certs/fullchain.pem
 WEB_SSL_PRIVKEY | The location of the SSL private key file | absolute filename | /etc/ssl/private/privkey.pem
+ASSETS_ENV | The assets environment assets are downloaded/applied from. If unset no asset actions will happen | a asset environment |
```
Comment (sroze) [inline php-nginx/README.md]: Can't we just have `ASSETS_PATH `? To me that looks like oversimplifying a bit and hiding a very simple thing which is `tools/assets/X`. So it'll be clear for everybody that we are un-tarring the archives at this location :)

Comment (andytson-inviqa) [inline php-nginx/README.md]: ASSETS_ENV controls the path in S3 too

Comment (andytson-inviqa) [inline php-nginx/README.md]: tbh, it'd make for a lot easier reading without ASSETS_ENV. This isn't much different to the magento one tho, which was done without ability to change the tools/assets bit

**Comment by andytson-inviqa** [comment]
ok, I've added the apache version in now this has been discussed

**Comment by andytson-inviqa** [comment]
there's no BC breakages in this PR, so merging shouldn't impact anything, but I've made changes, so should be reviewed again

**Pull request merged**